zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java

```diff
@@ -441,6 +452,7 @@ private void processAlerts(
             for (int alertId : v) {
                 RecordAlert recAlert = tableAlert.read(alertId);
                 Alert alert = new Alert(recAlert);
+                alert.setTags(tableAlertTag.getTagsByAlertId(alertId));
```

Comment:
This worries me. We know that this end point has performance issues and an extra call will only make it worse.
Especially as we make this call for every alert, not just the ones we're going to include.
If possible we should definitely only set the tags if we're going to include the alert. I'm also wondering if we should make the tags optional and only include them if requesed?